content/doc/book/pipeline/shared-libraries.adoc

```diff
@@ -218,204 +218,173 @@ this variable to checkout the appropriate version of the library.
+Use a deeper, unique directory structure to avoid resource name conflicts with other libraries.
 
-This approach has limitations; for example, it prevents the declaration of a
-superclass.
+=== Global variables
 
-Alternately, a set of `steps` can be passed explicitly to a library class, in a
-constructor, or just one method:
+If a file in the `vars` directory contains one or more methods (other than `call`),
+a global variable of that name is created, and the methods are added to that global.
+This is a convenient way to create a group of related methods in a
+single `.groovy` file and access as members of an object.
 
+For example:
+
+.var/log.groovy
 [source,groovy]
 ----
-package org.foo
-class Utilities implements Serializable {
-  def steps
-  Utilities(steps) {this.steps = steps}
-  def mvn(args) {
-    steps.sh "${steps.tool 'Maven'}/bin/mvn -o ${args}"
-  }
+def info(message) {
+    echo "INFO: ${message}"
 }
-----
-
-When saving state on classes, such as above, the class *must* implement the
-`Serializable` interface. This ensures that a Pipeline using the class, as seen
-in the example below, can properly suspend and resume in Jenkins.
 
-[source,groovy]
-----
-@Library('utils') import org.foo.Utilities
-def utils = new Utilities(steps)
-node {
-  utils.mvn 'clean package'
+def warning(message) {
+    echo "WARNING: ${message}"
 }
 ----
 
-If the library needs to access global variables, such as `env`, those should be
-explicitly passed into the library classes, or methods, in a similar manner.
-
-Instead of passing numerous variables from the Scripted Pipeline into a library,
-
+.Jenkinsfile
 [source,groovy]
 ----
-package org.foo
-class Utilities {
-  static def mvn(script, args) {
-    script.sh "${script.tool 'Maven'}/bin/mvn -s ${script.env.HOME}/jenkins.xml -o ${args}"
-  }
-}
+@Library('utils') _
+
+log.info 'Starting'
+log.warning 'Nothing to do!'
 ----
 
-The above example shows the script being passed in to one `static` method,
-invoked from a Scripted Pipeline as follows:
+Declarative Pipeline does not allow usage of this kind of global variable
+outside of a `script` directive
+(link:https://issues.jenkins-ci.org/browse/JENKINS-42360[JENKINS-42360]).
 
+.Jenkinsfile
 [source,groovy]
 ----
-@Library('utils') import static org.foo.Utilities.*
-node {
-  mvn this, 'clean package'
+@Library('utils') _
+
+pipeline {
+    agent none
+    stage ('Example') {
+        steps {
+             script { // <1>
+                 log.info 'Starting'
+                 log.warning 'Nothing to do!'
+             }
+        }
+    }
 }
 ----
+<1> `script` directive required to access global variables in Declarative Pipeline.
 
+[NOTE]
+====
+A global variable defined in a shared library will only show up in _Global Variables
+Reference_ (under _Pipeline Syntax_) after Jenkins loads and uses that library
+as part of a successful Pipeline run.
+====
 
-=== Defining global variables
+.Avoid preserving state in global variables
```

Comment:
I am not so happy about the following "avoid" formulations and maybe especially for the example demonstrating something that should not be used. Maybe just keep a single avoid sentence like: "Avoid preserving state in global variables[: ]Use a static class or instantiate a local variable of a class instead."? And delete the not recommended and thus kind of bad example completely?